frontend/src/concepts/pipelines/content/tables/pipelineVersion/PipelineVersionTable.tsx

```diff
@@ -0,0 +1,100 @@
+import * as React from 'react';
+import { TableVariant } from '@patternfly/react-table';
+import { TableBase } from '~/components/table';
+import { PipelineVersionKF } from '~/concepts/pipelines/kfTypes';
+import { pipelineVersionColumns } from '~/concepts/pipelines/content/tables/columns';
+import { useCheckboxTable } from '~/components/table';
+import PipelineVersionTableRow from '~/concepts/pipelines/content/tables/pipelineVersion/PipelineVersionTableRow';
+import { usePipelineVersionLoadMore } from '~/concepts/pipelines/content/tables/usePipelineLoadMore';
+import PipelineViewMoreFooterRow from '~/concepts/pipelines/content/tables/PipelineViewMoreFooterRow';
+
+type PipelineVersionTableProps = {
+  pipelineId: string;
+  initialVersions: PipelineVersionKF[];
+  nextPageToken?: string;
+  loading?: boolean;
+  totalSize: number;
+  sortField?: string;
+  sortDirection?: 'asc' | 'desc';
+  setSortField: (field: string) => void;
+  setSortDirection: (dir: 'asc' | 'desc') => void;
+  pipelineVersionDetailsPath: (namespace: string, id: string) => string;
+};
+
+const PipelineVersionTable: React.FC<PipelineVersionTableProps> = ({
+  pipelineId,
+  initialVersions,
+  nextPageToken,
+  loading,
+  totalSize,
+  sortField,
+  sortDirection,
+  setSortField,
+  setSortDirection,
+  pipelineVersionDetailsPath,
+}) => {
+  const { data: versions, onLoadMore } = usePipelineVersionLoadMore({
+    pipelineId,
+    initialData: initialVersions,
+    initialPageToken: nextPageToken,
+    sortDirection,
+    sortField,
+    loaded: !loading,
+  });
+  const { tableProps, toggleSelection, isSelected } = useCheckboxTable(
+    versions.map(({ id }) => id),
+  );
+
+  return (
+    <>
+      <TableBase
+        {...tableProps}
+        loading={loading}
+        itemCount={totalSize}
+        data={versions}
+        columns={pipelineVersionColumns}
+        rowRenderer={(version) => (
+          <PipelineVersionTableRow
+            key={version.id}
+            isChecked={isSelected(version.id)}
+            onToggleCheck={() => toggleSelection(version.id)}
+            version={version}
+            pipelineVersionDetailsPath={pipelineVersionDetailsPath}
```

Comment:
Is it necessary to pass along this prop? Is there another use case where the link path will differ? If not just generate the link path inside `PipelineVersionTableRow`

Reply:
I think the projects details page also needs this, let me double check it, maybe I forgot to take care of that page.